zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1858,24 +1923,47 @@ public Object call() throws Exception {
     });
   }
 
+  private boolean watchForData(final String path, boolean skipWatchingNodeNotExist) {
+    try {
+      if (skipWatchingNodeNotExist) {
+        Stat stat = new Stat();
+        retryUntilConnected(() -> (((ZkConnection) getConnection()).getZookeeper().getData(path, true, stat)));
```

Comment:
I think we did not discuss the performance impact here, now most of the time we are using getData() instead of exist(), which could increase read traffic.

Reply:
You are right, we can discuss this a little bit more in a meeting. It would be idea if Zookeeper has an exist() API that won't not install watch if the znode does not exist.